src/Provider/BaseProvider.php

```diff
@@ -153,12 +154,16 @@ public function requireThumbnails(): bool
 
     public function generateThumbnails(MediaInterface $media): void
     {
-        $this->thumbnail->generate($this, $media);
+        if ($this->thumbnail instanceof GeneratableThumbnailInterface) {
```

Comment:
I don't know enough this code to know if we have to do it this way or throw an exception instead. Do you know ?

Reply:
I think it is better this way. This code could be called by subclasses or this own class, for each media saved.